handshake_manager.go

```diff
@@ -103,6 +103,8 @@ func (c *HandshakeManager) handleOutbound(vpnIP uint32, f EncWriter, lighthouseT
 	if err != nil {
 		return
 	}
+	hostinfo.Lock()
```

Comment:
It could get confusing but at this point we are close to optimizing away the hostmap lock here